marshal/tinyint/marshal_utils.go

```diff
@@ -0,0 +1,205 @@
+package tinyint
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gocql/gocql/marshal"
+	"math"
+	"math/big"
+	"reflect"
+	"strconv"
+)
+
+var (
+	errMarshal = errors.Join(marshal.ErrorMarshal, errors.New("tinyint"))
```

Comment:
I just realized that you may actually avoid extra error check if you do wrapping like this on `tinyint` package level.
So, can you please remove `marshal.ErrorMarshal` from here, and wrap errors in `gocql.Marhsal` level.
But keep `errMarshal` in.

```suggestion
	errMarshal = errors.New("failed to marshal tinyint")
```

Reply:
Done